flows/runs/run.go

```diff
@@ -188,9 +190,17 @@ func (r *flowRun) ApplyEvent(s flows.Step, action flows.Action, event flows.Even
 		r.Session().LogEvent(s, action, event)
 	}
 
-	// eventEnvelope, _ := utils.EnvelopeFromTyped(event)
-	// eventJSON, _ := json.Marshal(eventEnvelope)
-	// fmt.Printf("⚡︎ in run %s: %s\n", r.UUID(), string(eventJSON))
+	if log.GetLevel() >= log.DebugLevel {
+		var origin string
+		if event.FromCaller() {
+			origin = "caller"
+		} else {
+			origin = "engine"
+		}
+		eventEnvelope, _ := utils.EnvelopeFromTyped(event)
+		eventJSON, _ := json.Marshal(eventEnvelope)
+		log.WithField("code", event.Type()).WithField("run", r.UUID()).WithField("data", string(eventJSON)).Debugf("%s event applied", origin)
```

Comment:
frustratingly, logrus doesn't give you control over field order, so if want a useful field to appear leftmost... its gotta be first A-Z

Reply:
also no way to prevent escaping of quotes (e.g. `"{\"type\"...`)... ezlog?